ios/samples/build_common.sh

```diff
@@ -1,56 +1,44 @@
 #!/usr/bin/env bash
 
-# set -e: Terminate script if a command returns an error
-set -e
-# set -u: Terminate script if an unset variable is used
-set -u
-# set -x: Debugging use, print each statement
-# set -x
-
 ## START STANDARD BUILD SCRIPT INCLUDE
 # adjust relative paths as necessary
 THIS_SCRIPT="$(readlink -f "${BASH_SOURCE[0]}")"
-. "${THIS_SCRIPT%/*}/../../resources/build/build-utils.sh"
+. "${THIS_SCRIPT%/*/*}/../resources/build/build-utils.sh"
 ## END STANDARD BUILD SCRIPT INCLUDE
 
+# TODO:  resolve how this is sourced by scripts in subfolders;
+# we want to cd to the subfolder including this script.
+#
+# # This script runs from its own folder
+# cd "$(dirname "$THIS_SCRIPT")"
+
+# Include our resource functions; they're pretty useful!
+. "$KEYMAN_ROOT/resources/shellHelperFunctions.sh"
+
 # Please note that this build script (understandably) assumes that it is running on Mac OS X.
-if [[ "${OSTYPE}" == *"darwin" ]]; then
-  echo "This build script will only run in a Mac environment."
-  exit 1
-fi
+verify_on_mac
 
 if [ -z "$TARGET" ]; then
   exit 1
 fi
 
-display_usage ( ) {
-    echo "build.sh [-no-update] | [-lib-build] | [-lib-ignore] | [-clean]"
-    echo
-    echo "  -clean          Removes all previously-existing build products for this project before building."
-    echo "  -no-update      If an in-place copy of KeymanEngine.xcframework exists, does not seek out an updated copy."
-    echo "  -lib-build      Actively rebuilds KMEI before copying its build products to project resources."
-    echo "  -lib-nobuild    Prevents the build script from building KeymanEngine under any circumstances."
-    echo "  -no-codesign    Performs the build without code signing."
-    echo "  -debug          Sets the configuration to debug mode instead of release."
-    echo
-    echo "  If no settings are specified this script will grab a copy of the most recent build of KeymanEngine,"
-    echo "  performing an initial build of it if necessary."
-    exit 1
-}
+builder_describe "Builds sample app $TARGET that demos the Keyman Engine for iPhone and iPad" \
+  "@/ios/engine build" \
+  "clean" \
+  "configure" \
+  "build" \
+  "--debug          Avoids codesigning and adds full sourcemaps for the embedded predictive-text engine"
 
-assert ( ) {
-    if ! [ -f $1 ]; then
-        echo "Build failed:  missing $1"
-        exit 1
-    fi
-}
+builder_parse "$@"
 
-verify_KMEI ( ) {
-    KMEI_BUILD_EXISTS=true
-    [ -d "$KEYMAN_ENGINE_FRAMEWORK_SRC" ] || KMEI_BUILD_EXISTS=false
-}
+CONFIG=Release
+if builder_is_debug_build; then
+  CONFIG="Debug"
+fi
```

Comment:
I see these have `Release` and `Debug` vs the engine and keyman definitions of `release` and `debug`? Is this correct?